# VarSCAT-analysis
## Scripts of biological analysis of Wang et al. VarSCAT: A computational tool for sequence context annotations of genomic variants.<br/> ## 
### The data used in manuscript are: <br/> ###
* Platinum Genome, hg38 small variant of NA12877 and NA12878 (https://www.illumina.com/platinumgenomes.html, https://github.com/Illumina/PlatinumGenomes/blob/master/files/2017-1.0.files)<br/>
* Genome in a Bottle, GRCh38 small variants in more difficult regions v4.2.1 of HG002-HG007 (https://www.nist.gov/programs-projects/genome-bottle, https://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release/)<br/>
* The 2548 human individual biallelic SNP+INDEL GRCh38 call sets from 1000 Genomes Porject (https://www.internationalgenome.org/, http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/release/20190312_biallelic_SNV_and_INDEL/)
* ClinVar database GRCh38 (https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/weekly/, date: 2022/01/09)<br/>
#### The source code and document of VarSCAT are available at: https://github.com/elolab/VarSCAT ####

#### The source code of VarSCAT, the analysis codes and processed data of manuscriptare  also available at https://doi.org/10.5281/zenodo.7018364/ ####
